code/items/equipments/entities/C4Entity.cs

```diff
@@ -53,45 +104,111 @@ public override void Simulate(Client cl)
         {
             if (IsClient)
             {
-
                 if (CreatedDisplay)
                 {
                     //No way to parent a world panel to an entity :(
                     //We need to find a better way of doing this.
                     //I heard supposedly you can use FrameSimulate, but I was getting weird duplicated results (think Windows XP window glitch)
-                    BombDisplay.Transform = GetAttachment("timer") ?? Transform;
-                    BombDisplay.WorldScale = 0.1f;
+                    TimerDisplay.Transform = GetAttachment("timer") ?? Transform;
+                    TimerDisplay.WorldScale = 0.05f;
 
                     return;
                 }
                 else
                 {
-                    BombDisplay = new WorldPanel();
+                    TimerDisplay = new WorldPanel();
                     CreatedDisplay = true;
 
-                    BombDisplay.AddClass("c4display");
-                    BombDisplay.StyleSheet.Add(StyleSheet.FromFile("/ui/alivehud/c4/c4Display.scss"));
+                    TimerDisplay.AddClass("c4worldtimer");
+                    TimerDisplay.StyleSheet.Add(StyleSheet.FromFile("/ui/alivehud/c4/C4WorldTimer.scss"));
+
+                    TimerDisplayLabel = TimerDisplay.AddChild<Label>();
+                    TimerDisplayLabel.Text = "--:--";
+                }
+            }
 
-                    var label = BombDisplay.AddChild<Label>();
-                    label.Text = "00:00";
+            // If the player moves away from the bomb or the bomb becomes armed, close the UI
+            if (_currentUsers.Count > 0 || State == C4State.Armed)
+            {
+                for (int i = _currentUsers.Count - 1; i >= 0; i--)
+                {
+                    if (Vector3.DistanceBetween(_currentUsers[i].Controller.Position, Position) > 100)
+                    {
+                        _currentUsers[i].ClientCloseC4Menu();
+                        _currentUsers.Remove(_currentUsers[i]);
+                    }
```

Comment:
Why not handle it clientside?

Reply:
Yeah, that would be better. I'm still not sure how the client/server side code works.
I can move this logic to the C4ArmPanel